GlassBR - plateLen, plateWidth, charWeight, pbTol, and tNT inserted twice in SymbolMap

All of these UIDs are inserted twice into the GlassBR `SymbolMap`. I grouped them together in one issue here because they all seem to stem from the same problem. See #4062 for where the UIDs are added specifically.

Unlike #4062 however, this is not an instance of a UID being added twice from two different locations, rather it is from the same location but being referenced twice.

In `symbolsForTable`, there are two lists added to it: `inputs` and `inputDataConstraints`.
https://github.com/JacquesCarette/Drasil/blob/0ab5dec39cad09265bc57ad36805d980f6337296/code/drasil-example/glassbr/lib/Drasil/GlassBR/Symbols.hs#L16-L19

Both of these lists contain the contents of two other lists: `inputsWUnitsUncrtn` and `inputsWUncrtn`. This is where the duplicate UIDs lie, since `inputs` and `inputDataConstraints` both contain these lists, their subsequent contents are added twice in `symbolsForTable` as well.

`inputs`:
https://github.com/JacquesCarette/Drasil/blob/0ab5dec39cad09265bc57ad36805d980f6337296/code/drasil-example/glassbr/lib/Drasil/GlassBR/Unitals.hs#L44-L46

`inputDataConstraints`:
https://github.com/JacquesCarette/Drasil/blob/0ab5dec39cad09265bc57ad36805d980f6337296/code/drasil-example/glassbr/lib/Drasil/GlassBR/Unitals.hs#L68-L71

While in both of these functions the contents are mapped using different functions, their UIDs remain the same causing the issue. Similar to #4062, determining which duplicate should be removed.

I tried simply removing either of the terms from `symbolsForTable` however that seems to remove some necessary UIDs as well, as I get these errors:

**Removing** `inputs ++ `**:**

Resulting code:
```haskell
symbolsForTable :: [QuantityDict]
symbolsForTable = outputs ++ tmSymbols ++ map qw specParamVals ++ 
  map qw symbolsWithDefns ++ map qw symbols ++ map qw unitless ++
  map qw inputDataConstraints ++ interps
```

Output:
```console
glassbr.EXE: Symbol: nomThick not found in SymbolMap
CallStack (from HasCallStack):
  error, called at lib\Database\Drasil\ChunkDB.hs:116:24 in drasil-database-0.1.1.0-4jdMY7TWows9S8CmX9loPI:Database.Drasil.ChunkDB
make: *** [Makefile:311: glassbr_gen] Error 1
```

**Removing** ` ++ map qw inputDataConstraints`**:**

Resulting code:
```haskell
symbolsForTable :: [QuantityDict]
symbolsForTable = inputs ++ outputs ++ tmSymbols ++ map qw specParamVals ++ 
  map qw symbolsWithDefns ++ map qw symbols ++ map qw unitless ++ interps
```

Output:
```console
glassbr.EXE: Symbol: aR not found in SymbolMap
CallStack (from HasCallStack):
  error, called at lib\Database\Drasil\ChunkDB.hs:116:24 in drasil-database-0.1.1.0-4jdMY7TWows9S8CmX9loPI:Database.Drasil.ChunkDB
make: *** [Makefile:311: glassbr_gen] Error 1
```

Not really sure where to go from here, maybe changing one of the UIDs?

--- a/code/drasil-example/glassbr/lib/Drasil/GlassBR/Body.hs
+++ b/code/drasil-example/glassbr/lib/Drasil/GlassBR/Body.hs
@@ -44,9 +44,9 @@ import Drasil.GlassBR.Requirements (funcReqs, inReqDesc, funcReqsTables, nonfunc
 import Drasil.GlassBR.Symbols (symbolsForTable, thisSymbols)
 import Drasil.GlassBR.TMods (tMods)
 import Drasil.GlassBR.Unitals (blast, blastTy, bomb, explosion, constants,
-  constrained, inputDataConstraints, inputs, outputs, specParamVals, glassTy,
+  constrained, inputs, outputs, specParamVals, glassTy,
   glassTypes, glBreakage, lateralLoad, load, loadTypes, pbTol, probBr, stressDistFac, probBreak,
-  sD, termsWithAccDefn, termsWithDefsOnly, terms)
+  sD, termsWithAccDefn, termsWithDefsOnly, terms, dataConstraints)
 
 srs :: Document
 srs = mkDoc mkSRS (S.forGen titleize phrase) si
@@ -107,7 +107,7 @@ mkSRS = [TableOfContents,
         , GDs [] [] HideDerivation -- No Gen Defs for GlassBR
         , DDs [] ([Label, Symbol, Units] ++ stdFields) ShowDerivation
         , IMs [instModIntro] ([Label, Input, Output, InConstraints, OutConstraints] ++ stdFields) HideDerivation
-        , Constraints auxSpecSent inputDataConstraints
+        , Constraints auxSpecSent dataConstraints
         , CorrSolnPpties [probBr, stressDistFac] []
         ]
       ],

--- a/code/drasil-example/glassbr/lib/Drasil/GlassBR/Symbols.hs
+++ b/code/drasil-example/glassbr/lib/Drasil/GlassBR/Symbols.hs
@@ -8,15 +8,15 @@ import Theory.Drasil (output)
 
 import Drasil.GlassBR.IMods (iMods)
 import Drasil.GlassBR.ModuleDefs (allMods, implVars)
-import Drasil.GlassBR.Unitals (inputDataConstraints, inputs, outputs, 
-  specParamVals, symbols, symbolsWithDefns, unitless, tmSymbols, interps)
+import Drasil.GlassBR.Unitals (inputDataConstraints, inputs, outputs,
+  specParamVals, symbols, symbolsWithDefns, unitless, tmSymbols, interps, derivedInputDataConstraints)
 
 import Data.List ((\\))
 
 symbolsForTable :: [QuantityDict]
 symbolsForTable = inputs ++ outputs ++ tmSymbols ++ map qw specParamVals ++ 
   map qw symbolsWithDefns ++ map qw symbols ++ map qw unitless ++
-  map qw inputDataConstraints ++ interps
+  map qw derivedInputDataConstraints ++ interps
 
   -- include all module functions as symbols
 thisSymbols :: [QuantityDict]

--- a/code/drasil-example/glassbr/lib/Drasil/GlassBR/Unitals.hs
+++ b/code/drasil-example/glassbr/lib/Drasil/GlassBR/Unitals.hs
@@ -31,8 +31,9 @@ modElas = uc' "modElas" (nounPhraseSP "modulus of elasticity of glass")
 {--}
 
 constrained :: [ConstrainedChunk]
-constrained = map cnstrw inputDataConstraints ++
-  [cnstrw probBr, cnstrw probFail, cnstrw stressDistFac, cnstrw nomThick, cnstrw glassTypeCon]
+constrained = map cnstrw dataConstraints ++ [nomThick, cnstrw glassTypeCon]
+ -- map cnstrw inputDataConstraints ++ map cnstrw derivedInputDataConstraints -- ++
+  -- [cnstrw probBr, cnstrw probFail, cnstrw stressDistFac, cnstrw nomThick, cnstrw glassTypeCon]
 
 plateLen, plateWidth, aspectRatio, charWeight, standOffDist :: UncertQ
 pbTol, tNT :: UncertainChunk
@@ -49,26 +50,33 @@ inputs = map qw inputsWUnitsUncrtn ++ map qw inputsWUncrtn ++
 inputsWUnitsUncrtn :: [UncertQ]
 inputsWUnitsUncrtn = [plateLen, plateWidth, charWeight]
 
---derived inputs with units and uncertainties
-derivedInsWUnitsUncrtn :: [UncertQ]
-derivedInsWUnitsUncrtn = [standOffDist]
-
 --inputs with uncertainties and no units
 inputsWUncrtn :: [UncertainChunk]
 inputsWUncrtn = [pbTol, tNT]
-
---derived inputs with uncertainties and no units
-derivedInsWUncrtn :: [UncertQ]
-derivedInsWUncrtn = [aspectRatio]
 
 --inputs with no uncertainties
 inputsNoUncrtn :: [ConstrainedChunk]
 inputsNoUncrtn = [cnstrw glassTypeCon, nomThick]
 
+--derived inputs with units and uncertainties
+derivedInsWUnitsUncrtn :: [UncertQ]
+derivedInsWUnitsUncrtn = [standOffDist]
+
+--derived inputs with uncertainties and no units
+derivedInsWUncrtn :: [UncertQ]
+derivedInsWUncrtn = [aspectRatio]
+
 inputDataConstraints :: [UncertainChunk]
 inputDataConstraints = map uncrtnw inputsWUnitsUncrtn ++
-  map uncrtnw inputsWUncrtn ++ map uncrtnw derivedInsWUnitsUncrtn ++
-  map uncrtnw derivedInsWUncrtn
+  map uncrtnw inputsWUncrtn
+
+derivedInputDataConstraints :: [UncertainChunk]
+derivedInputDataConstraints = map uncrtnw derivedInsWUnitsUncrtn
+  ++ map uncrtnw derivedInsWUncrtn
+
+dataConstraints :: [UncertainChunk]
+dataConstraints = inputDataConstraints ++ derivedInputDataConstraints
+
 
 plateLen = uqcND "plateLen" (nounPhraseSP "plate length (long dimension)")
   lA metre Real

